src/witan/gateway/components/query_router.clj

```diff
@@ -0,0 +1,41 @@
+(ns witan.gateway.components.query-router
+  (:require [com.stuartsierra.component :as component]
+            [taoensso.timbre            :as log]
+            [ring.util.http-response    :refer [ok not-found failed-dependency]]
+            [witan.gateway.protocols    :as p :refer [RouteQuery]]
+            [clj-http.client            :as client]))
+
+
+
+(defrecord QueryRouter [service-map]
+  RouteQuery
+  (route-query [this route params]
+    (let [[nsp query] ((juxt (comp keyword namespace) name) route)]
+      (if (contains? service-map nsp)
+        (let [{:keys [host port]} (get service-map nsp)
+              url (str "http://" host ":" port "/" query)]
+          (log/debug "Routing query to" url params)
+          (try
+            (let [{:keys [status body headers] :as response} (client/get url {:query-params params})]
+              (when-not (= status 200)
+                (log/warn "Query returned an odd status code:" response))
```

Comment:
we'll probably want to improve the exception handling here too. It might be worth looking at something like Urania. Either using or stealing ideas from: https://github.com/funcool/urania